solidity-plugin/src/main/java/org/sonarsource/solidity/SoliditySensor.java

```diff
@@ -82,25 +83,55 @@ public void analyzeFiles(SensorContext context, List<InputFile> files) {
     for (InputFile file : files) {
       String lastAnalyzedFile = file.toString();
       if (inSonarQube(context)) {
-        LOG.debug("Analyzing: " + lastAnalyzedFile);
-        getSyntaxHighlighting(context, file).save();
+        try {
+          LOG.debug("Analyzing: " + lastAnalyzedFile);
+          SolidityParser parser = Utils.returnParserUnitFromParsedFile(file.contents());
+          getSyntaxHighlighting(parser, context, file).save();
+          saveFileMeasures(context, computeMeasures(parser, fileLinesContextFactory.createFor(file), file), file);
+          if (inSonarQube(context)) {
+
+          }
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
       } else {
         LOG.debug(lastAnalyzedFile);
       }
     }
   }
 
-  public NewHighlighting getSyntaxHighlighting(SensorContext context, InputFile inputFile) {
-    this.highlighting = context.newHighlighting().onFile(inputFile);
-    try {
-      SolidityParser parser = Utils.returnParserUnitFromParsedFile(inputFile.contents());
-      MyVisitor visitor = new MyVisitor(this.highlighting);
-      visitor.visitTokens(parser.getTokenStream());
-      visitor.highlightComments(parser);
-    } catch (IOException e) {
-      LOG.error(e.toString());
-    }
-    return this.highlighting;
+  private FileMeasures computeMeasures(SolidityParser parser, FileLinesContext fileLinesContext, InputFile file) {
+    MetricsVisitor metricsVisitor = new MetricsVisitor(parser);
+
+    return metricsVisitor.fileMeasures;
+  }
+
+  public NewHighlighting getSyntaxHighlighting(SolidityParser parser, SensorContext context, InputFile inputFile) {
+    NewHighlighting highlighting = context.newHighlighting().onFile(inputFile);
+    MyVisitor visitor = new MyVisitor(highlighting);
+    visitor.visitTokens(parser.getTokenStream());
+    visitor.highlightComments(parser);
+    return highlighting;
+  }
+
+  private static void saveFileMeasures(SensorContext context, FileMeasures fileMeasures, InputFile inputFile) {
+    System.out.println("CCComputed: " + fileMeasures);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this use of System.out or System.err by a logger. ([squid:S106](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS106))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=16&issues=AWQTGK75bXXTsx7O4tIT&open=AWQTGK75bXXTsx7O4tIT)

[ISSUE_KEY]: AWQTGK75bXXTsx7O4tIT